File: tests/07_phases_omp.dg/common/fortran/success_parallel_02.f90
! <testinfo>
! test_generator=config/mercurium-omp
! test_nolink=yes
! </testinfo>

MODULE M
    IMPLICIT NONE
    CONTAINS
        SUBROUTINE FOO(V, N)
            IMPLICIT NONE
            INTEGER :: N
            INTEGER :: V(N, *)
        END SUBROUTINE FOO

        SUBROUTINE BAR(V, N)
            IMPLICIT NONE
            INTEGER :: N
            INTEGER :: V(N, *)
            !$OMP PARALLEL
            CALL FOO(V, N)
            !$OMP END PARALLEL
        END SUBROUTINE BAR

END MODULE M

File: tests/07_phases_omp.dg/iomp/c/success_parallel_02.c
/*
<testinfo>
test_generator=config/mercurium-iomp
</testinfo>
*/

/*This tests shared/private variables*/

#include <assert.h>

int omp_get_num_threads(void);
int omp_get_thread_num(void);

int main(void) {
    int a = -1;
    #pragma omp parallel num_threads(4) firstprivate(a)
    {
        a++;
    }
    assert(a == -1);

    #pragma omp parallel num_threads(4)
    {
        a++;
    }
    assert(a != -1);
}
